### PR TITLE
Allowing all parameters in the Service Discovery request

### PR DESCRIPTION
**Pull request by srijitamukherjee**
Also added the integration test for service_discovery/auth_controller.

Closes [THREESCALE-4851](https://issues.redhat.com/browse/THREESCALE-4851)

### CODE REVIEW

File: app/controllers/master/service_discovery/auth_controller.rb [hunk 1]
```diff
@@ -11,7 +11,7 @@ def show
   protected
 
   def self_domain_url(self_domain)
-    options = params.except(:self_domain, :scope).merge(host: self_domain, controller: 'provider/admin/service_discovery/auth')
+    options = params.except(:self_domain, :scope).merge(host: self_domain, controller: 'provider/admin/service_discovery/auth').permit!
```
Comment (guicassolato) [inline app/controllers/master/service_discovery/auth_controller.rb:14]: This is a bit risky. Have you tried instead...

```suggestion
    options = params.permit(*%i[code referrer state]).merge(host: self_domain, controller: 'provider/admin/service_discovery/auth')
```
?

File: app/controllers/master/service_discovery/auth_controller.rb [hunk 2]
```diff
@@ -11,7 +11,7 @@ def show
   protected
 
   def self_domain_url(self_domain)
-    options = params.except(:self_domain, :scope).merge(host: self_domain, controller: 'provider/admin/service_discovery/auth')
+    options = params.permit(:code, :referrer, :state).merge(host: self_domain, controller: 'provider/admin/service_discovery/auth', action: :show)
```
Comment (hallelujah) [inline app/controllers/master/service_discovery/auth_controller.rb:14]:
```suggestion
    options = params.permit(:code, :referrer, :state).merge(host: self_domain, controller: controller_path, action: :show)
```

Comment (hallelujah) [reply app/controllers/master/service_discovery/auth_controller.rb:14]: ⚠️  Nothing to do with this PR but the `self_domain` is suggested to a security issue. It is fetched from the request params.

Comment (srijitamukherjee) [reply app/controllers/master/service_discovery/auth_controller.rb:14]: ack, will change, thanks

**Pull request merged**